update/interface.go

```diff
@@ -0,0 +1,89 @@
+package update
+
+import (
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+)
+
+type StateSyncer interface {
+	GetMetaBlock() *block.MetaBlock
+	SyncAllState(epoch uint32) error
+	GetAllTries() (map[string]data.Trie, error)
+	GetAllTransactions() (map[string]data.TransactionHandler, error)
+	GetAllMiniBlocks() (map[string]*block.MiniBlock, error)
+	IsInterfaceNil() bool
+}
+
+// TrieSyncer synchronizes the trie, asking on the network for the missing nodes
+type TrieSyncer interface {
+	StartSyncing(rootHash []byte) error
+	Trie() data.Trie
+	IsInterfaceNil() bool
+}
+
+// AccountsHandlerContainer keep a list of AccountsAdapters
+type AccountsHandlerContainer interface {
+	Get(key string) (state.AccountsAdapter, error)
+	Add(key string, val state.AccountsAdapter) error
+	AddMultiple(keys []string, interceptors []state.AccountsAdapter) error
+	Replace(key string, val state.AccountsAdapter) error
+	Remove(key string)
+	Len() int
+	IsInterfaceNil() bool
+}
+
+// TrieSyncerContainer keep a list of TrieSyncer
```

Comment:
Typo in comment, also add for the rest of the interfaces that are missing comments